sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor{TPartition}.cs

```diff
@@ -1066,12 +1075,144 @@ async Task performProcessing()
+                        partitionIds = default;
+                    }
+
+                    // Perform a load balancing cycle only in the case that partitions were successfully read.
+
+                    if (partitionIds != default)
+                    {
+                        var claimedOwnership = default(EventProcessorPartitionOwnership);
+
+                        try
+                        {
+                            claimedOwnership = await LoadBalancer.RunLoadBalancingAsync(partitionIds, cancellationToken).ConfigureAwait(false);
+                        }
+                        catch (EventHubsException ex)
+                            when (Resources.OperationClaimOwnership.Equals(ex.GetFailureOperation(), StringComparison.InvariantCultureIgnoreCase))
+                        {
+                            // If a specific partition was associated with the failure, it is carried by a well-known data member of the exception.
+
+                            var partitionId = ex.GetFailureData<string>();
+
+                            var partition = (partitionId ?? string.Empty) switch
+                            {
+                                string id when (id == string.Empty) => null,
+                                string _ when (ActivePartitionProcessors.TryGetValue(partitionId, out var partitionProcessor)) => partitionProcessor.Partition,
+                                _ => new TPartition { PartitionId = partitionId }
+                            };
+
+                            _ = InvokeOnProcessingErrorAsync(ex.InnerException ?? ex, partition, Resources.OperationClaimOwnership, CancellationToken.None);
+                            Logger.EventProcessorClaimOwnershipError(Identifier, EventHubName, ConsumerGroup, partitionId, ((ex.InnerException ?? ex).Message));
+                        }
+                        catch (Exception ex) when (ex.IsNotType<CancellationToken>())
+                        {
+                            _ = InvokeOnProcessingErrorAsync(ex, null, Resources.OperationLoadBalancing, CancellationToken.None);
+                            Logger.EventProcessorLoadBalancingError(Identifier, EventHubName, ConsumerGroup, ex.Message);
+                        }
+
+                        // If a partition was claimed, begin processing it.
+
+                        if (claimedOwnership != default)
+                        {
+                            await StartProcessingPartitionAsync(claimedOwnership.PartitionId, cancellationToken).ConfigureAwait(false);
+                        }
+                    }
+
+                    // Some ownership for some previously claimed partitions may have expired or have been stolen; stop the processing for partitions
+                    // which are no longer owned.
+
+                    cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+
+                    await Task.WhenAll(ActivePartitionProcessors.Keys
```

Comment:
Full test coverage for the blocks on 1162-1165 and 1173 - 1188 fit more naturally with the Start/Stop partition context, so I've deferred them for now.